lib/AST/ASTPrinter.cpp

```diff
@@ -2679,11 +2679,14 @@ void PrintAST::visitConstructorDecl(ConstructorDecl *decl) {
 
   if ((decl->getInitKind() == CtorInitializerKind::Convenience ||
        decl->getInitKind() == CtorInitializerKind::ConvenienceFactory) &&
-      !decl->getAttrs().hasAttribute<ConvenienceAttr>()) {
+      !decl->getAttrs().hasAttribute<ConvenienceAttr>() &&
+      !decl->getDeclContext()->getAsProtocolExtensionContext()) {
```

Comment:
How about `class`? Inherited `init` from protocol extension in `class` should be printed as `convenience`.
IIRC, we don't have test case for that. https://github.com/apple/swift/pull/11099#issuecomment-317303609

Reply:
Hm, I guess so. Okay, I'll check the effective context, or whatever it's called.